### ISSUE TITLE
Using `exit` in a module's function prematurely exits the PowerShell session.

### ISSUE THREAD

**Issue by SetTrend** [PowerShell/PowerShell#21379]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://learn.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

Calling PowerShell from another shell, it is necessary to **report exit codes** to the other shell (which is a requirement for, e.g., [`git` hooks](https://github.com/git-for-windows/git/issues/4870#issuecomment-2002504789)).

Doing this from an advanced function declared in a PowerShell module, prematurely exits the function – and even the current PowerShell session – without returning the exit code.

See this [Question at StackOverflow](https://stackoverflow.com/questions/78227176/how-can-i-return-an-exit-code-from-a-powershell-modules-function-without-exitin) for an MRE.

### Expected behavior

```console
PowerShell should return the exit code.
```

### Actual behavior

```console
No exit code is returned. The PowerShell session ends prematurely.
```

### Environment data

```powershell
Name                           Value
----                           -----
PSVersion                      7.4.1
PSEdition                      Core
GitCommitId                    7.4.1
OS                             Microsoft Windows 10.0.19045
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by rhubarb-geek-nz** [comment]
System.Environment.Exit works on both Linux and Windows

```
$ cat exit.ps1
#!/usr/bin/env pwsh

[System.Environment]::Exit(42)
$ ./exit.ps1
$ echo $?
42
```

or

```
D:\tmp>type exit.ps1
#!/usr/bin/env pwsh

[System.Environment]::Exit(42)

D:\tmp>pwsh exit.ps1

D:\tmp>echo %ERRORLEVEL%
42
```

or

```
D:\tmp>pwsh
PowerShell 7.4.1
PS D:\tmp> pwsh exit.ps1
PS D:\tmp> $LastExitCode
42
```

**Comment by jhoneill** [comment]
You might want to read the later part of this 
https://jhoneill.github.io/powershell/2019/08/21/ExitThrowReturnEtc.html 

Exit is no better understood now than when I wrote that, but depending on how launch the script you can send an exit code back to PowerShell which then says "OK I've run all the commands I was told to run, I'm exiting normally"

**Comment by SetTrend** [comment]
Thanks, @rhubarb-geek-nz and @jhoneill, for clarifying! 👍 

@jhoneill, I found your article very enlightening on what happens when using `exit`.

So, to have my script run according to PowerShell paradigm and using it from within Bash, I decided to return `$true` and `$false`, resp. in order to avoid the exception output. In Bash, I'll convert that return value to the PowerShell session's exit code then:

```bash
pwsh -Command "Import-Module 'C:\Temp\Test-Exit.psm1; exit (Text-Exit '') ? 0 : 1'"
```
